backend/app/schemas/exam.py

```
@@ -103,6 +113,61 @@ def _normalize_optional_url(value: Optional[str]) -> Optional[str]:
 
+def _validate_options_for_type(options: List[str], item_type: Optional[str]) -> List[str]:
+    """Enforce per-item-type option shape.
+
+    A `field_validator` on `options` rather than a model validator, so the error
+    keeps its historic `loc` of `["body", "options"]`. `type` is declared before
+    `options` on every request model, so it is already validated and readable
+    from `info.data`.
+
+    `multiple-choice` — the default, and the only type today's frontend sends —
+    keeps the exact-4 rule it has always had. Only explicitly non-default types
+    see the relaxed 2–8 range or the no-options text/essay shape.
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)